File: DragonsGenerator.WEB/src/app/core/models/Creatures/creature.ts
export interface CreatureAbility {
  score: number;
  modifier: string;
}

export interface CreatureNamedEntry {
  name: string;
  description: string;
}

export interface Creature {
  id: string;
  name: string;
  category: string;
  part: string | null;
  section: string | null;
  type: string;
  armorClass: number;
  armorNote: string | null;
  hitPoints: string;
  woundThreshold: number | null;
  speed: string;
  abilities: Record<string, CreatureAbility>;
  savingThrows: string | null;
  skills: string | null;
  senses: string | null;
  languages: string | null;
  challengeRating: string;
  xp: number;
  traits: CreatureNamedEntry[];
  actions: CreatureNamedEntry[];
  reactions: CreatureNamedEntry[];
  legendaryActions: CreatureNamedEntry[];
  description: string;
}

File: DragonsGenerator.API/Models/Creature.cs
namespace DragonsGenerator.API.Models;

public record CreatureAbility(int Score, string Modifier);

public record CreatureNamedEntry(string Name, string Description);

public record Creature(
    string Id,
    string Name,
    string Category,
    string? Part,
    string? Section,
    string Type,
    int ArmorClass,
    string? ArmorNote,
    string HitPoints,
    int? WoundThreshold,
    string Speed,
    Dictionary<string, CreatureAbility> Abilities,
    string? SavingThrows,
    string? Skills,
    string? Senses,
    string? Languages,
    string ChallengeRating,
    int Xp,
    List<CreatureNamedEntry> Traits,
    List<CreatureNamedEntry> Actions,
    List<CreatureNamedEntry> Reactions,
    List<CreatureNamedEntry> LegendaryActions,
    string Description
);
